lib/fizzy/execute.cpp

```diff
@@ -358,64 +400,159 @@ execution_result execute(Instance& instance, FuncIdx function, std::vector<uint6
             instance.globals[idx] = stack.pop();
             break;
         }
-        // FIXME: make this into a template?
         case Instr::i32_load:
         {
-            const auto address = static_cast<uint32_t>(stack.pop());
-            // NOTE: alignment is dropped by the parser
-            const auto offset = read<uint32_t>(immediates);
-            if ((address + offset + sizeof(uint32_t)) > instance.memory.size())
+            if (!load_from_memory<uint32_t, uint32_t>(instance, stack, immediates))
```

Comment:
In the declaration of the template you can do `template <typename DstT, typename SrcT = DstT>` and omit one type here: `load_from_memory<uint32_t>`.